### ISSUE TITLE
Convert an object into a type without executing a query

### ISSUE THREAD

**Issue by crpahl** [rmosolgo/graphql-ruby#4599]
**Is your feature request related to a problem? Please describe.**

There are situations where we would like to serialize an object as a type without executing an actual query.

**Describe the solution you'd like**

For example, instead of executing:

```
SomeSchema.execute(...)
```

We would like to be able to convert an object directly a type:

```
SomeType.wrap(some_object, some_context)
```

Is that the most reasonable way to do this? Taking this approach requires us creating our own `GraphQL::Query::Context` which doesn't feel quite right. We also don't necessarily need the context and also explored using `GraphQL::Query::NullContext` which didn't work either sadly.

Any thoughts/suggestions here would be greatly appreciated!

**Comment by rmosolgo** [comment]
This definitely strikes me as a "design smell" -- why do you want to make a GraphQL type instance outside of GraphQL? 

I'm hesitant to support it because I don't think it should happen. If there's complex logic inside the GraphQL type, it should be extracted (to a "lower" layer) for reuse or testing. But maybe you have something in mind that I haven't run into before...?

**Comment by andrewjdmccann** [comment]
Hey @rmosolgo,

@crpahl asked me to type up a response for this question! I definitely agree that this is generally an anti pattern and could understand if you decide not to support it but I'll explain the use case and you can make that call.

We're in a transition scenario right now where we're migrating from an old Rails application to a new react powered GraphQL API.

The reason we were looking to make this change is to act as a bridge between the two systems while we're making that transition. During this transition window we will have some parts of our application rendered server side via React components, but those components will be fed properties that come from Rails rather than through our GraphQL API.

We want those objects that are being fed into these server rendered components to be as close to what we would see from a GraphQL API as possible to make the final transition to the GraphQL API as simple as possible. The problem is right now they need to serialize these objects into the GraphQL Types but there's no *good* way to do it.

We could introduce another layer that creates models that closely resemble what our GraphQL output would end up being and then just serialize that, however, to get those models to look very close to GraphQL we would need to implement our own GraphQL-like type system, that includes things like Enums, Connection Types, and all of the `GraphQL::Schema::Scalar` types. Additionally we've implemented permission checks on our types at the GraphQL Type layer that I'd also want to enfoce, and would need to move to this layer.

TL;DR: It is a "design smell" but it's one we need to do for pragmatic reasons while we execute a strategy to migrate our code base from a Rails Monolith to a GraphQL/React powered page. The ability to split the GraphQL serialization layer from the GraphQL query will allow us to transition to a much cleaner GraphQL structure long term.

**Comment by rmosolgo** [comment]
I hope that's helpful! I don't have plans to work further on this so I'm going to close it. Let me know if you give it a try and run into trouble :+1:
